### ISSUE TITLE
Multiset Result and Records are attached to derived configuration, not original one

### ISSUE THREAD

**Issue by yamass** [jOOQ/jOOQ#17184]
### Expected behavior

When executing a query with the `multiset` operator, the records in `multiset(...).convertFrom(r -> ...)` (in this case the `r` record) should be attached to the same configuration as the `DslContext` executing the query.

### Actual behavior

Those records have a completely different (default?) configuration:

![jooq-multiset-config](https://github.com/user-attachments/assets/efff344e-0718-41e6-92c7-83149f15cdd0)

In our case, this means that we can `store()` changed main records of the query, but not sub-records (joined with `multiset`). 

Also, when trying to store them, we don't get any meaningful exception but 

```
Exception in thread "main" org.jooq.exception.DataAccessException: SQL [insert into "postgres"."b" ("name") values (?) returning "postgres"."b"."id"]; Connection is closed
	at org.jooq_3.19.11.POSTGRES.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3609)
	at org.jooq.impl.Tools.translate(Tools.java:3597)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:827)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:362)
	at org.jooq.impl.TableRecordImpl.storeInsert0(TableRecordImpl.java:207)
	at org.jooq.impl.TableRecordImpl.lambda$storeInsert$0(TableRecordImpl.java:172)
	at org.jooq.impl.RecordDelegate.operate(RecordDelegate.java:144)
	at org.jooq.impl.TableRecordImpl.storeInsert(TableRecordImpl.java:171)
	at org.jooq.impl.UpdatableRecordImpl.store0(UpdatableRecordImpl.java:228)
	at org.jooq.impl.UpdatableRecordImpl.lambda$store$0(UpdatableRecordImpl.java:155)
	at org.jooq.impl.RecordDelegate.operate(RecordDelegate.java:144)
	at org.jooq.impl.UpdatableRecordImpl.store(UpdatableRecordImpl.java:154)
	at org.jooq.impl.UpdatableRecordImpl.store(UpdatableRecordImpl.java:146)
	at de.yamass.db.Main.main(Main.java:53)
Caused by: java.sql.SQLException: Connection is closed
	at com.zaxxer.hikari.pool.ProxyConnection$ClosedConnection.lambda$getClosedConnection$0(ProxyConnection.java:503)
	at jdk.proxy3/jdk.proxy3.$Proxy8.prepareStatement(Unknown Source)
	at com.zaxxer.hikari.pool.ProxyConnection.prepareStatement(ProxyConnection.java:328)
	at com.zaxxer.hikari.pool.HikariProxyConnection.prepareStatement(HikariProxyConnection.java)
	at org.jooq.impl.ProviderEnabledConnection.prepareStatement(ProviderEnabledConnection.java:109)
	at org.jooq.impl.SettingsEnabledConnection.prepareStatement(SettingsEnabledConnection.java:91)
	at org.jooq.impl.ProviderEnabledConnection.prepareStatement(ProviderEnabledConnection.java:109)
	at org.jooq.impl.SettingsEnabledConnection.prepareStatement(SettingsEnabledConnection.java:91)
	at org.jooq.impl.AbstractQuery.prepare(AbstractQuery.java:431)
	at org.jooq.impl.AbstractDMLQuery.prepare0(AbstractDMLQuery.java:975)
	at org.jooq.impl.AbstractDMLQuery.prepare(AbstractDMLQuery.java:959)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:315)
	... 10 more
```

### Steps to reproduce the problem

```sql
create table A
(
    id   int GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
    name text
);

create table B
(
    id   int GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
    a_id int not null references A,
    name text
);

insert into A (name) values ('blah');
insert into B (a_id, name) values ((select id from a), 'blah');
```

```java
DefaultConfiguration jooqConfig = new DefaultConfiguration();
DataSourceConnectionProvider connectionProvider = new DataSourceConnectionProvider(dataSource);
jooqConfig.setConnectionProvider(connectionProvider);
jooqConfig.setSQLDialect(SQLDialect.POSTGRES);
jooqConfig.set(new ThreadLocalTransactionProvider(connectionProvider));
```

```java
System.out.println(dslContext.configuration());

List<BRecord> bs = dslContext.select(A)
		.select(multiset(DSL.select(B)
				.from(B)
				.where(B.A_ID.eq(A.ID)))
				.as("bs").convertFrom(r -> r.map(record -> {
							System.out.println(record.configuration());
							return record.into(B);
						})
				))
		.from(A)
		.<List<BRecord>>fetchOne(record -> {
			System.out.println(record.configuration());
			return record.get("bs", List.class);
		});

BRecord firstB = bs.get(0);
firstB.setName("something else");
firstB.store();
```

### jOOQ Version

3.19.11

### Database product and version

Postgres 16

### Java Version

21

### JDBC / R2DBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.7.2

**Comment by lukaseder** [comment]
Fixed in versions:

- 3.20.0
- 3.19.12 (https://github.com/jOOQ/jOOQ/issues/17191)
- 3.18.19 (https://github.com/jOOQ/jOOQ/issues/17192)
- 3.17.28 (https://github.com/jOOQ/jOOQ/issues/17193)

**Comment by lukaseder** [comment]
Note, as a workaround, you can re-attach records to your original configuration, of course, using `Record.attach(Configuration)`

**Comment by yamass** [comment]
Yes, we were aware of that workaround.

Thanks for the quick fix! Looking forward to 1.19.12.
